### PR TITLE
Synchronize access to wordnet

### LINKED ISSUE #81: Multithreading issues with nltk tokenizer
The nltk-wordnet library is causing issues when multiple threads call whoami. See http://stackoverflow.com/questions/27433370/what-would-cause-wordnetcorpusreader-to-have-no-attribute-lazycorpusloader for a way to solve this problem.

Sample stack trace:

```
GenerateWorker-1 - ERROR - 'WordNetCorpusReader' object has no attribute '_LazyCorpusLoader__args'
Traceback (most recent call last):
  File "./generate_symbols_and_synonyms.py", line 108, in insert_types
    lowercase=True, get_rep=False))
  File "/scratch//.virtualenvs/pywb/src/whoami/whoami/query.py", line 43, in get_assertion_definitions
    entry = extract_assertion(assertion, subject=subject)
  File "/scratch//.virtualenvs/pywb/src/whoami/whoami/defextract.py", line 184, in extract_assertion
    postprocess_entry(entry, substitutions)
  File "/scratch//.virtualenvs/pywb/src/whoami/whoami/postprocess.py", line 211, in postprocess_entry
    raise_norp(entry)
  File "/scratch//.virtualenvs/pywb/src/whoami/whoami/postprocess.py", line 105, in raise_norp
    not wtype.is_person(definition):
  File "/scratch//.virtualenvs/pywb/src/whoami/whoami/wordtypes.py", line 47, in is_person
    person_ss = wn.synset('person.n.01')
  File "/scratch//.virtualenvs/pywb/local/lib/python2.7/site-packages/nltk/corpus/util.py", line 99, in __getattr__
    self.__load()
  File "/scratch//.virtualenvs/pywb/local/lib/python2.7/site-packages/nltk/corpus/util.py", line 73, in __load
    args, kwargs  = self.__args, self.__kwargs
```

### PR DESCRIPTION
**Pull request by moldot**
Use threading.Lock to guard access to wordnet object in wordtypes.py.
This should eliminate the error thrown by wordnet when accessed by
multiple threads.

Fix #81

### CODE REVIEW

File: whoami/wordtypes.py [hunk 1]
```diff
@@ -44,11 +48,14 @@ def is_person(node):
     if not spos.is_noun(node):
         return False
     try:
+        wn_lock.acquire()
```
Comment (alvaromorales) [inline whoami/wordtypes.py]: It's better to grab the lock in a `with` statement.

```python
with wn_lock:
    try:
        person_ss = wn.synset('person.n.01')
        ...
    except IndexError:
        return False
return True
```

See http://stackoverflow.com/questions/29373448/what-is-the-optimal-use-of-a-lock-with-a-try-except-in-python-2-7/29374011

File: scripts/test_threads.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+try:
+    t1 = MyThread(u'James is a runner and a computer scientist')
+    t2 = MyThread(u'James is a monk in Massachusetts')
+    t3 = MyThread(u'James is the president of the national rifle')
+
+    t1.start()
+    t2.start()
+    t3.start()
+except:
+    print 'BOOM'
```
Comment (alvaromorales) [inline scripts/test_threads.py]: If you were to not use the `wn_lock`, would this script raise an exception?

Comment (moldot) [inline scripts/test_threads.py]: Yes, the same exception posted in #81

**Comment by alvaromorales** [comment]
lgtm :rocket:

**Comment by alvaromorales** [comment]
Still getting this error:

```
GenerateWorker-11 - ERROR - 'WordNetCorpusReader' object has no attribute '_LazyCorpusLoader__args'
Traceback (most recent call last):
  File "./generate_symbols_and_synonyms.py", line 109, in insert_types
    lowercase=True, get_rep=False))
  File "/scratch//whoami/whoami/query.py", line 43, in get_assertion_definitions
    entry = extract_assertion(assertion, subject=subject)
  File "/scratch//whoami/whoami/defextract.py", line 184, in extract_assertion
    postprocess_entry(entry, substitutions)
  File "/scratch//whoami/whoami/postprocess.py", line 211, in postprocess_entry
    raise_norp(entry)
  File "/scratch//whoami/whoami/postprocess.py", line 105, in raise_norp
    not wtype.is_person(definition):
  File "/scratch//whoami/whoami/wordtypes.py", line 52, in is_person
    person_ss = wn.synset('person.n.01')
  File "/scratch//.virtualenvs/pywb/local/lib/python2.7/site-packages/nltk/corpus/util.py", line 99, in __getattr__
    self.__load()
  File "/scratch//.virtualenvs/pywb/local/lib/python2.7/site-packages/nltk/corpus/util.py", line 73, in __load
    args, kwargs  = self.__args, self.__kwargs
AttributeError: 'WordNetCorpusReader' object has no attribute '_LazyCorpusLoader__args'
GenerateWorker-11 - DEBUG -   .. whoami exception for: "A relish is a cooked, pickled, or chopped vegetable or fruit food item typically used as a condiment in particular to enhance a staple." ([u'relish'
, u'Relish'])
```

**Comment by moldot** [comment]
I'll try to replicate and see what's the problem.

**Pull request merged**